cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/query/QueryCoveringIndexesTest.java

```diff
@@ -115,6 +116,32 @@ public void returnsNullWhenQueryContainsInvalidValue() throws Exception {
     }
 
     @Test
+    public void returnsDocForQueryAgainstIndexWithSpecialChar() throws Exception {
+        DocumentRevision rev = new DocumentRevision("mike12");
+        Map<String, Object> bodyMap = new HashMap<String, Object>();
+        bodyMap.put("name", "mike");
+        bodyMap.put("age", 12);
+        bodyMap.put("pet", "cat");
+        rev.setBody(DocumentBodyFactory.create(bodyMap));
+        ds.createDocumentFromRevision(rev);
+        assertThat(im.ensureIndexed(Arrays.<Object>asList("name", "age"), "basic index"), is
+                ("basic index"));
```

Comment:
I don't think it is needed since the index creation tests all those paths to be sure, if a single special char works for index querying works eg space, then all the others should be successfully be escaped